runtime/src/bank/builtins/mod.rs

```diff
@@ -1,110 +1,98 @@
     },
     BuiltinPrototype {
-        feature_id: None,
+        enable_feature_id: None,
+        core_bpf_migration: None,
         program_id: solana_config_program::id(),
         name: "config_program",
         entrypoint: solana_config_program::config_processor::Entrypoint::vm,
     },
     BuiltinPrototype {
-        feature_id: None,
+        enable_feature_id: None,
+        core_bpf_migration: None,
         program_id: bpf_loader_deprecated::id(),
         name: "solana_bpf_loader_deprecated_program",
         entrypoint: solana_bpf_loader_program::Entrypoint::vm,
     },
     BuiltinPrototype {
-        feature_id: None,
+        enable_feature_id: None,
+        core_bpf_migration: None,
         program_id: bpf_loader::id(),
         name: "solana_bpf_loader_program",
         entrypoint: solana_bpf_loader_program::Entrypoint::vm,
     },
     BuiltinPrototype {
-        feature_id: None,
+        enable_feature_id: None,
+        core_bpf_migration: None,
         program_id: bpf_loader_upgradeable::id(),
         name: "solana_bpf_loader_upgradeable_program",
         entrypoint: solana_bpf_loader_program::Entrypoint::vm,
     },
     BuiltinPrototype {
-        feature_id: None,
+        enable_feature_id: None,
+        core_bpf_migration: None,
         program_id: solana_sdk::compute_budget::id(),
         name: "compute_budget_program",
         entrypoint: solana_compute_budget_program::Entrypoint::vm,
     },
     BuiltinPrototype {
-        feature_id: None,
+        enable_feature_id: None,
+        core_bpf_migration: None,
         program_id: solana_sdk::address_lookup_table::program::id(),
         name: "address_lookup_table_program",
         entrypoint: solana_address_lookup_table_program::processor::Entrypoint::vm,
     },
     BuiltinPrototype {
-        feature_id: Some(feature_set::zk_token_sdk_enabled::id()),
+        enable_feature_id: Some(feature_set::zk_token_sdk_enabled::id()),
+        core_bpf_migration: None,
         program_id: solana_zk_token_sdk::zk_token_proof_program::id(),
         name: "zk_token_proof_program",
         entrypoint: solana_zk_token_proof_program::Entrypoint::vm,
     },
     BuiltinPrototype {
-        feature_id: Some(feature_set::enable_program_runtime_v2_and_loader_v4::id()),
+        enable_feature_id: Some(feature_set::enable_program_runtime_v2_and_loader_v4::id()),
+        core_bpf_migration: None,
         program_id: solana_sdk::loader_v4::id(),
         name: "loader_v4",
         entrypoint: solana_loader_v4_program::Entrypoint::vm,
     },
 ];
+
+pub static EPHEMERAL_BUILTINS: &[EphemeralBuiltinPrototype] = &[
+    EphemeralBuiltinPrototype {
+        core_bpf_migration: None,
+        program_id: solana_sdk::feature::id(),
+        name: "feature_gate_program",
+    },
+    EphemeralBuiltinPrototype {
+        core_bpf_migration: None,
+        program_id: solana_sdk::native_loader::id(),
+        name: "native_loader_program",
+    },
```

Comment:
Yeah, it can always be added. This list does absolutely nothing but migrate programs, so if we decide to migrate the native loader one day, we'll just add it here.